### ISSUE TITLE
docs: al_open_directory(), al_create_fs_entry() needs a better clarification

### ISSUE THREAD

**Issue by gh0stwizard** [liballeg/allegro5#707]
Hi,
I've started using Allegro's filesystem overlay and found out that it is hard to understand at first glance that al_create_fs_entry() actually does NOT create any (i)node on the filesystem. What al_create_fs_entry(path) doing is:
* allocates memory for ALLEGRO_FS_ENTRY *ptr
* calls operating system's stat(path) function
* fills-in *ptr with the information retrieved from the stat() function
* returns ptr to the caller

And after that I may use al_open_directory(). This is differs from classic Unix documentation, such K & R, where says:
* DIR * opendir(char *dirname)
* Dirent *readdir(DIR *dfd) vs. 
* void closedir(DIR *dfd)

same in Allegro:
* opendir: 

1. ALLEGRO_FS_ENTRY *al_create_fs_entry(const char *path)
2. bool al_open_directory(ALLEGRO_FS_ENTRY *)

* readdir: ALLEGRO_FS_ENTRY *al_read_directory(ALLEGRO_FS_ENTRY *e)
* closedir: bool al_close_directory(ALLEGRO_FS_ENTRY *e)

My proposal is expand notes of al_open_directory() to say something like this: you may want to use al_create_fs_entry() before using this function. So, at least a developer will know "how to get this ALLEGRO_FS_ENTRY *entry" to pass it to al_open_directory().
And second, please, expand notes of al_create_fs_entry() by telling that this is an analogue of stat() function and no i-node will be created. (Yes, I know that there is al_update_fs_entry(), but anyway).

Thanks!

**Comment by elias-pschernig** [comment]
I agree, this is definitely missing. Right now al_open_directory does not even refer to al_create_fs_entry at all, not even a "See also". I think it should mention that you have to use that function to obtain your ALLEGRO_FS_ENTRY for the directory, add it to "See also", and then also a short example to list the files in a directory.

**Comment by elias-pschernig** [comment]
Maybe we even should have an al_open_directory_name() function, similar to how we have al_filename_exists() and all_fs_entry_exists().

**Comment by gh0stwizard** [comment]
Well, I think that a function similar to DIR *opendir(char *dirname) will be a good addition. For instance,

ALLEGRO_FS_ENTRY *al_open_directory_name(const char *path);

Returns a pointer to ALLEGRO_FS_ENTRY * on success. Otherwise, returns NULL and set an error by calling al_set_errno(). Because al_open_directory_name() may fail in next cases:
* the file does not exist on the filesystem
* not enough rights to open the directory
* an underhood check was failed, like buffer overflow

I.e. there is no need to return boolean value like the original al_open_directory() does.
